quill-sql/src/main/scala/io/getquill/context/sql/idiom/SqlIdiom.scala

```diff
@@ -86,6 +86,8 @@ trait SqlIdiom extends Idiom {
   }
 
   implicit def sqlQueryTokenizer(implicit strategy: NamingStrategy): Tokenizer[SqlQuery] = Tokenizer[SqlQuery] {
+    case FlattenSqlQuery(List(QueryContext(q, _)), None, Nil, Nil, None, None, List(SelectValue(i: Ident, None)), fase) =>
```

Comment:
We should avoid transformations here. Could you try to change `SqlQuery` to introduce this behavior?